packages/astro/test/custom-404-injected-from-dep.test.js

```diff
@@ -0,0 +1,22 @@
+import { expect } from 'chai';
+import { loadFixture } from './test-utils.js';
+
+describe('Custom 404 with injectRoute from dependency', () => {
```

Comment:
Added a specific test for #7397! The injected route will be matched to a relative path in `./node_modules` but the build should still succeed. This test verifies that this is the case.